runtime/metricsview/astsql.go

```diff
@@ -102,42 +116,60 @@ func (b *sqlBuilder) writeSelect(n *SelectNode) error {
 	if n.FromTable != nil {
 		b.out.WriteString(*n.FromTable)
 
-		// Add unnest joins. We only and always apply these against FromPlain (ensuring they are already unnested when referenced in outer SELECTs).
+		// Add unnest joins. We only and always apply these against FromTable (ensuring they are already unnested when referenced in outer SELECTs).
 		for _, u := range n.Unnests {
 			b.out.WriteString(", ")
 			b.out.WriteString(u)
 		}
 	} else if n.FromSelect != nil {
-		b.out.WriteByte('(')
-		err := b.writeSelect(n.FromSelect)
-		if err != nil {
-			return err
-		}
-		b.out.WriteString(") ")
-		b.out.WriteString(n.FromSelect.Alias)
-
-		for _, ljs := range n.LeftJoinSelects {
-			err := b.writeJoin("LEFT", n.FromSelect, ljs)
+		if n.FromSelect.IsCTE {
+			b.out.WriteString(n.FromSelect.Alias)
+			if n.JoinComparisonSelect == nil {
+				return fmt.Errorf("internal: FromSelect is a CTE but JoinComparisonSelect is nil, this is used only for comparison queries")
+			}
+			// find the CTE for FromSelect
+			var cte *SelectNode
+			for _, c := range n.CTEs {
+				if c.Alias == n.FromSelect.Alias {
+					cte = c
+					break
+				}
+			}
```

Comment:
Can't it just use `n.FromSelect` directly? Won't it still be the same pointer, just with `IsCTE == true`?